### PR TITLE
[Insets] Doc fixes

### LINKED ISSUE #366: Please mention usage of InsetAwareTopAppBar in docs
I just spend a huge chunk of my time trying to get the TopAppBar working correctly because it didn't draw behind the statusbar, like it would when using Insetter.

I just found out that the samples use a special version called InsetAwareTopAppBar (https://github.com/google/accompanist/blob/main/sample/src/main/java/com/google/accompanist/sample/insets/EdgeToEdgeLazyColumn.kt) to achieve the same behavior. Please mention it in the docs so that future readers don't have the same issue.

### LINKED ISSUE #369: There is an error in the code in the documentation
PIG3ONMBQ V40ZXY41WN}YD
UUJIN{S{6_A%9`_ D8~F2N9

here https://google.github.io/accompanist/insets/

### PR DESCRIPTION
**Pull request by chrisbanes**
Fixes #369
Fixes #366

### CODE REVIEW

File: docs/insets.md [hunk 1]
```diff
@@ -96,6 +101,20 @@ For a more complex example, see the [`EdgeToEdgeLazyColumn`](https://github.com/
...
+Unfortunately, most of Compose Material's layouts do not support the use of content padding, which means that the following code probably doesn't produce the effect you want:
+
+``` kotlin
+TopAppBar(
+    // content
+    modifier = Modifier.statusBarsPadding()
+)
+```
+
+To workaround this, our sample has a number of wrapper layouts in the [`InsetAwareLayouts.kt`](https://github.com/google/accompanist/blob/main/sample/src/main/java/com/google/accompanist/sample/insets/InsetAwareLayouts.kt)
```
Comment (ricknout) [inline docs/insets.md:115]: "the [`InsetAwareLayout.kt`] file"? Or remove "the"

File: sample/src/main/java/com/google/accompanist/sample/insets/InsetAwareLayouts.kt [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+ */
+@Composable
+fun InsetAwareTopAppBar(
+    title: @Composable () -> Unit,
+    modifier: Modifier = Modifier,
+    navigationIcon: @Composable (() -> Unit)? = null,
+    actions: @Composable RowScope.() -> Unit = {},
+    backgroundColor: Color = MaterialTheme.colors.primarySurface,
+    contentColor: Color = contentColorFor(backgroundColor),
+    elevation: Dp = 4.dp
```
Comment (ricknout) [inline sample/src/main/java/com/google/accompanist/sample/insets/InsetAwareLayouts.kt:47]: `AppBarDefaults.TopAppBarElevation`?

File: docs/insets.md (hunk 1 above)
Comment (chrisbanes) [reply docs/insets.md]: Added. Visual Studio didn't save the file with my finished sentence before uploading 👎

File: sample/src/main/java/com/google/accompanist/sample/insets/InsetAwareLayouts.kt (hunk 2 above)
Comment (chrisbanes) [reply sample/src/main/java/com/google/accompanist/sample/insets/InsetAwareLayouts.kt]: Done

**Review by ricknout — APPROVED** [review]

**Review by nickbutcher — APPROVED** [review]

**Pull request merged**